sdk/python/requirements.txt

```diff
@@ -1,19 +1,48 @@
-urllib3>=1.15,<1.25 #Fixing the version conflict with the "requests" package caused by the kubernetes package
-six>=1.10
-certifi
-python-dateutil
-PyYAML
-google-cloud-storage>=1.13.0
-kubernetes>=8.0.0, <=10.0.0
-PyJWT>=1.6.4
-cryptography>=2.4.2
-google-auth>=1.6.1
-requests_toolbelt>=0.8.0
-cloudpickle==1.1.1
-kfp-server-api==0.2.5
-argo-models == 2.2.1a
-jsonschema >= 3.0.1
-tabulate == 0.8.3
-click == 7.0
-Deprecated
-strip-hints
+#
+# This file is autogenerated by pip-compile
+# To update, run:
+#
+#    pip-compile --output-file=requirements.txt requirements.in
```

Comment:
Previously, we could only stably update requirements.txt using docker: https://github.com/kubeflow/pipelines/pull/3161/files#diff-4a38caa8c75b104ebd2a31a36f5c3b44R4

Looks like different python version will have different set of available dependencies, so generated requirements.txt can be different.
I'm not very familiar with which python versions our sdk needs to support, will this work for sdk?

Reply:
My impression was that for KFP the packages are the same regardless of the python version. Do you have a specific example of difference?
In Travis we test 2.5, 3.6, 3.7 and want to test 3.8.

Reply:
https://github.com/kubeflow/pipelines/pull/3161#issuecomment-590986792
The image was using python 3.5, probably it hasn't been supported by some new libraries.

Reply:
I found one package (zipp) that has dropped support for python3.5.
So it makes sense to create the snapshot using the lowest version of python that we support. The snapshot will be tested on all supported python version by the presubmit tests.

Reply:
SGTM